src/api/v1/profiles/handlers/update-profile.ts

```
@@ -0,0 +1,137 @@
+      data: preprocessedData,
+    });
+
+    res.json(updatedProfile);
+  } catch (error) {
+    console.error("Failed to update profile:", error);
+    if (error instanceof z.ZodError) {
+      const validationResult: ProfileValidationResponse = {
+        success: false,
+        errors: error.errors.reduce(
+          (acc, err) => ({
+            ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)